packages/main/src/plugin/troubleshooting.ts

```diff
@@ -0,0 +1,110 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import AdmZip from 'adm-zip';
+import moment from 'moment';
+import * as os from 'node:os';
+import * as fs from 'node:fs';
+
+export interface FileMap {
+  fileName: string;
+  content: string;
+}
+
+export type LogType = 'log' | 'warn' | 'trace' | 'debug' | 'error';
+
+export class Troubleshooting {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  constructor(private apiSender: any) {}
+
+  // Save a list of files with their filename and content to a zip file
+  async saveLogsToZip(fileMaps: FileMap[], destination: string): Promise<void> {
+    const zip = new AdmZip();
+
+    // Add files to the zip
+    fileMaps.forEach(fileMap => {
+      zip.addFile(`${fileMap.fileName}.txt`, Buffer.from(fileMap.content, 'utf8'));
+    });
+
+    // Write the zip file to the destination
+
+    // Only write if fileMaps is not empty
+    if (fileMaps.length > 0) {
+      zip.writeZip(destination);
+      console.log(`Zip file saved to ${destination}`);
+    }
+  }
+
+  getConsoleLogs(consoleLogs: { logType: LogType; message: string }[]): FileMap[] {
+    const content = consoleLogs.map(log => `${log.logType} : ${log.message}`).join('\n');
+    return [
+      {
+        fileName: generateLogFileName('console'),
+        content: content,
+      },
+    ];
+  }
+
+  // TODO: Implement any other logs (Windows & Linux) we wish to retrieve in the future.
+  async getSystemLogs(): Promise<FileMap[]> {
+    let logs: FileMap[] = [];
+    if (os.platform() === 'darwin') {
+      logs = await this.getMacSystemLogs();
+    } else if (os.platform() === 'win32') {
+      logs = await this.getWindowsSystemLogs();
+    } else {
+      console.log('Unsupported platform for retrieving system logs');
+    }
+    return logs;
+  }
+
+  // Retrieve the mac system log files from ~/Library/Logs/Podman Desktop
+  private async getMacSystemLogs(): Promise<FileMap[]> {
+    const logs: FileMap[] = [];
+    const logFiles = ['launchd-stdout', 'launchd-stderr'];
+    const logDir = `${os.homedir()}/Library/Logs/Podman Desktop`;
+    for (const file of logFiles) {
+      try {
+        const filePath = `${logDir}/${file}.log`;
+        const content = await fs.promises.readFile(filePath, 'utf8');
+        logs.push({ fileName: generateLogFileName('system-' + file), content });
+      } catch (error) {
+        console.error(`Error reading ${file}.log: `, error);
+      }
+    }
+    return logs;
+  }
+  // Retrieve the windows system log files from ~/AppData/Roaming/Podman Desktop/logs/podman-desktop.log
```

Comment:
This would become a generic `getFileslog(files: string[])`
And we would pass for mac:
```
${os.homedir()}/Library/Logs/Podman Desktop/launchd-stdout
${os.homedir()}/Library/Logs/Podman Desktop/launchd-stderr
```

for windows:
```
${os.homedir()}/AppData/Roaming/Podman Desktop/logs/podman-desktop
```